packages/pass-style/src/types.d.ts

```diff
@@ -0,0 +1,182 @@
+ * and can therefore be safely shared with subgraphs that should not be able
+ * to communicate with each other.
+ * Without that assumption, such a guarantee requires a marshal-unmarshal round
+ * trip (as exists between vats) to produce data structures disconnected from
+ * any potential proxies.
+ */
+export type PureData = Passable<never, never>;
+export type TaggedRecord<
+  S extends string,
+  I extends InterfaceSpec,
+> = PassStyled<S> & {
+  [Symbol.toStringTag]: I;
+};
+/**
+ * An object marked as remotely accessible using the `Far` or `Remotable`
+ * functions, or a local presence representing such a remote object.
+ *
+ * A more natural name would be Remotable, but that could be confused with the
+ * value of the `Remotable` export of this module (a function).
+ */
+export type RemotableObject<I extends string = string> = TaggedRecord<
+  'remotable',
+  I
+>;
+/**
+ * The authority-bearing leaves of a Passable's pass-by-copy superstructure.
+ */
+export type PassableCap = Promise<any> | RemotableObject;
+/**
+ * A Passable sequence of Passable values.
+ */
+export type CopyArray<T extends Passable = any> = Array<T>;
+
+/**
+ * A Passable dictionary in which each key is a string and each value is Passable.
+ */
+export type CopyRecord<T extends Passable = any> = Record<string, T>;
+/**
+ * A Passable "tagged record" with semantics specific to the tag identified in
+ * the `[Symbol.toStringTag]` property (such as 'copySet', 'copyBag',
+ * or 'copyMap').
+ * It must have a property with key equal to the `PASS_STYLE` export and
+ * value 'tagged'
+ * and no other properties except `[Symbol.toStringTag]` and `payload`.
+ */
+export type CopyTagged<
+  Tag extends string = string,
+  Payload extends Passable = any,
+> = TaggedRecord<'tagged', Tag> & {
+  payload: Payload;
+};
+/**
+ * This is an interface specification.
+ * For now, it is just a string, but will eventually be `PureData`. Either
+ * way, it must remain pure, so that it can be safely shared by subgraphs that
+ * are not supposed to be able to communicate.
+ */
+export type InterfaceSpec = string;
+/**
+ * Internal to a useful pattern for writing checking logic
+ * (a "checkFoo" function) that can be used to implement a predicate
+ * (an "isFoo" function) or a validator (an "assertFoo" function).
+ *
+ *  * A predicate ideally only returns `true` or `false` and rarely throws.
+ *  * A validator throws an informative diagnostic when the predicate
+ *    would have returned `false`, and simply returns `undefined` normally
+ *    when the predicate would have returned `true`.
+ *  * The internal checking function that they share is parameterized by a
+ *    `Checker` that determines how to proceed with a failure condition.
+ *    Predicates pass in an identity function as checker. Validators
+ *    pass in `assertChecker` which is a trivial wrapper around `assert`.
+ *
+ * See the various uses for good examples.
+ */
+export type Checker = (
```

Comment:
No change suggested.

Just FYI as of https://github.com/endojs/endo/pull/1935 I expect I'll migrate `Checker` to a lower level utils package. But there's no reason for that to affect this PR, which will merge first.